playbooks/roles/logstash/templates/logstash.conf.j2

```diff
@@ -1,54 +1,129 @@
+    }
+  } else if [type] == "rabbitmq" {
+    grok {
+      match => { "message" => "%{WORD}\=%{QUOTEDSTRING:syslog_timestamp} %{WORD}\=%{QUOTEDSTRING:rabbitmq_vhost} %{WORD}\=%{QUOTEDSTRING:rabbitmq_queue} %{WORD}\=%{INT:rabbitmq_length}" }
     }
+  } else if [type] == "mongo" {
+    grok {
+      match => { "message" => "%{TIMESTAMP_ISO8601:syslog_timestamp} ?\[?%{DATA:mongo_context}?\]? %{GREEDYDATA:@message}" }
+    } 
+  } else if [type] == "tracking" { 
+    grok {
+      match => { "message" => "%{GREEDYDATA:@message}" }
+    }
+  } else if [type] == "logstash" {
+    grok {
+      match => { "message" => "\{\:%{WORD}\=\>\"%{TIMESTAMP_ISO8601:syslog_timestamp}\"\, %{GREEDYDATA:@message}\}" }
+    }
+  } else if [type] == "elasticsearch" {
+    grok {
+      match => { "message" => "(?:\[%{TIMESTAMP_ISO8601:syslog_timestamp}\]\[%{WORD:log_level}? \])%{GREEDYDATA:@message}" }
+    }
+  } else if [type] == "rsyslog" {
+    grok {
+      match => { "message" => "%{SYSLOGTIMESTAMP:syslog_timestamp} %{WORD:syslog_hostname} %{DATA:syslog_program}(?:\[%{POSINT:syslog_pid}\])?: %{GREEDYDATA:@message}" }
+      match => { "message" => "%{SYSLOGTIMESTAMP:syslog_timestamp} %{WORD:syslog_hostname} \[%{WORD}\=%{DATA:service_variant}\]\[%{DATA:service_program}\]\[%{WORD}\:%{USERNAME:env}\] %{WORD:log_level} \[%{WORD} ? %{POSINT:syslog_pid}\] \[%{DATA:filename}\:%{INT:line_number}\] \- %{GREEDYDATA:@message}" }
+      tag_on_failure => []
+    }
+    syslog_pri { }
     # Try and parse the tracking log json
     # 142 is syslog facility 17 (local1) and Informational.
     # This is used to reduce the number of errors in json parsing as
     # tracking uses that facility and priority by default.
     if "142" in [syslog_pri] {
       json {
-        source => "syslog_message"
+        source => "@message"
         target => "tracking"
       }
     }
-    if !("_grokparsefailure" in [tags]) {
-      mutate {
-        replace => [ "@source_host", "%{syslog_hostname}" ]
-        replace => [ "@message", "%{syslog_message}" ]
-      }
+  } else if ("beats_input_raw_event" in [tags]) {
+    mutate {
+      add_tag => [ "topbeat" ]
+      remove_tag => [ "beats_input_raw_event" ]
+    }
+  } else {
+    mutate {
+      add_tag => [ "unknown" ]
+    }
+  }
+  date {
+    match => [ "syslog_timestamp", "MMM  d HH:mm:ss", "MMM dd HH:mm:ss", "DD MMM YYYY:HH:mm:ss ZZZ", "ISO8601"]
+  }
+  if !("_grokparsefailure" in [tags]) {
+    mutate {
+      replace => [ "@source_host", "%{host}" ]
     }
     mutate {
-      remove_field => [ "syslog_hostname", "syslog_message", "syslog_timestamp" ]
+      remove_field => [ "syslog_hostname", "syslog_timestamp" ]
+      remove_tag => [ "beats_input_codec_plain_applied" ]
     }
   }
+  }
 }
 
 output {
-  # Example just to output to elasticsearch
```

Comment:
It seemed like an unnecessary/inaccurate comment. This is not an example configuration, and it is clear elastic search is an output